compiler/src/lnntoamm/Expr.ts

```diff
@@ -84,15 +87,31 @@ export default abstract class Expr {
           expr = call;
           ii += 1;
           skipDotIfNext();
-        } else if (next.has('methodsep')) {
+        } else if (expr !== null) {
+          // it's a field access
+          if (!(expr instanceof Ref)) {
+            const dec = Dec.gen(expr, metadata);
+            generated.push(dec);
+            expr = dec.ref();
+          }
+          // ensure that the value has the field
+          let fieldTy = Type.generate();
+          const hasField = Type.hasField(varName, fieldTy);
+          if (!expr.ty.compatibleWithConstraint(hasField, metadata.scope)) {
+            throw new Error(`cannot access ${varName} on type ${expr.ty.name} because it doesn't have that field`);
+          }
+          expr.ty.constrain(hasField, metadata.scope);
+          // TODO: better ast
```

Comment:
Can you explain what's wrong with the current AST here and what you'd like to change?

Reply:
the current ast only provides the name of the field - it'd be better to also see what the field is accessing (so for example, `getThing().field` would result in the `AccessField`'s AST just being `field`, whereas it'd be nice to have the full part.) It'll take quite a bit more logic to get it working and I have to provide a better AST in operator calls anyways so it'll have to be a future PR. I'll update the comment with that info though